client/blocks/get-apps/apps-badge.jsx

```diff
@@ -130,4 +146,4 @@ export class AppsBadge extends PureComponent {
 
 export default connect( null, {
 	recordTracksEvent,
-} )( localize( AppsBadge ) );
```

Comment:
Sorry, I just realized we still need this after verifying we have the [side effects I raised up before](https://github.com/Automattic/wp-calypso/pull/42507#discussion_r429149694) (my bad 🙇‍♂️).

Once we bring back the `localize` HOC, we'd need to also make sure `titleText` and `altText` are returned by a function so they can take into account the current locale. 

You can either use your initial approach (passing `translate` as argument: `getTitleText: ( translate ) => translate( ... )`) or the imported `translate` (`getTitleText: () => translate( ... )`).

Otherwise, we'll be using a wrong locale in the translations. Note how the attributes are still in English after switching my locale to Spanish:

<img width="1127" alt="Screen Shot 2020-05-22 at 11 51 05" src="https://user-images.githubusercontent.com/1233880/82656766-91f91f80-9c24-11ea-85cf-c218c289b9f1.png">

Reply:
I might just get fancy and use the `useTranslate` hook. I'll work on that later 👍

Reply:
Using the hook wasn't of much help as the APP_STORE_BADGE_URLS are outside and I would have had to pass `translate`  by function `getTitleText`, which by itself would solve the problem.
I just crated the `getTitleText` and `getAltText` function. But I couldn't test: could you tell me how you changed the locale without having having the component rendered? To change the language, I had tyo navigate in the settings, which would unmount the component anyway....

Reply:
> could you tell me how you changed the locale without having having the component rendered?

I simply navigated to Me > Get apps, then to My Profile, updated my locale and navigated back to Get apps. 

I'm not familiar with the React component lifecycle, but when the translations are static (meaning they are set as constants outside of a component), they seem to don't change even if the component is unmounted and mounted again.